extension/src/project/openWebHelp.ts

```diff
@@ -0,0 +1,174 @@
+import * as vscode from 'vscode';
+import { webHelpContainer } from "../completion/autocompletionProvider"; // Currently where all ./data/ files are being loaded & was maintained for symmetry
+
+
+// Triggers VSCode to open the official web documentation URL related to the currently selected symbol
+// Note: this function is directly referenced by projectCommands.ts and the the package.json contributes (commands & menus) section.
+export function openWebHelp() {
+	const editor: vscode.TextEditor = vscode.window.activeTextEditor;
+	let selected: string = editor.document.getText(editor.selection);	
+	let urlPath: string = webHelpContainer.getWebHelpUrlBySymbolName(selected);
+	vscode.env.openExternal(vscode.Uri.parse(urlPath));
+}
+
+
+// This container object represents all the normalized data extracted from help.autodesk.com/view/OARX/
+export class WebHelpLibrary{
```

Comment:
How does this extraction process work? Did you do it manually or do you have some automated process? I'm concerned about the maintenance effort. We (at least I) don't know how often these values (especially the guids) change.

Reply:
So the standalone C# app that I have doing this work is a 2 step process that I currently have divided into 3 steps for efficiency and testing purposes. 

1. The left-side navigation system does actually have the links until they've been expanded and no amount of inline tweaking to the DOM exposes the link; IE, they must be clicked. So, first I just go expand all the things containing links I want to extract and export the HTML of that navigation system/delete anything I don't want as an extraction target. This is technically the only part of the process required to obtain guids because their links use them.

2. At this point the application (with minor tweaks) could go scrape from a URL or it can download the contents of the first <article> it finds to a local file. I've opted to download because it takes time to load pages before the DOM has all the required content. IE, the pages are more like templates and the server has to lookup what should be dynamically pushed into different parts of the template.

3. Scraping the document (regardless of source) is somewhat well structured. It doesn't have what I'd call perfect identifiers, but the first <h1> is **always** the title, the first <pre> is **always** the signature, etcetera. So, you can very easily identify sections and design rules around how those should be evaluated/stored in my custom object abstraction. Now, the content itself can get very subjective do to documentation inconsistencies and I don't know if I want to try explaining that in a comment, but that is why I have been saying I need to further develop the logic to extract good type data; which isn't currently being used, but structurally exists in the JSON object.

**GUID**
I am not worried at all about the guid system. Autodesk has maintained these since 2016 and across all languages. The only way these change is if they completely redesign their help system.

**Content**
Like the GUIDs, the actual contents of these help files hasn't really changed since 2016. The behaviors of these functions hasn't really changed since 2000! Granted there are some exceptions where commands were added, but those are far and few between; much more likely that if something is introduced it just won't be documented...

Reply:
We need to bring this C# app or its typescript/node equivalent into this repo so that anybody can run it and update the JSON.

Reply:
I completely agree and am committed to doing exactly that, but I need to do a **lot** of cleanup on it before I make it publicly available. It is safe to say it was an experiment that needs to be pull together in a fluid way and the above steps more formally documented first.

**Note**
I don't think I am remotely familiar enough ts/node to convert this in any kind of a timely manner. However, I do think that would be an excellent exercise for my own personal growth. If you want me to go that direction, then please don't hold up finalizing this PR over it.